File: python/tools/dashboards/delete.py
import json

from schema.tool_inputs import DashboardDeleteSchema
from tools.types import Context, TextContent, Tool, ToolResult


async def delete_dashboard_handler(context: Context, params: DashboardDeleteSchema) -> ToolResult:
    project_id = await context.get_project_id()
    result = await context.api.dashboards(project_id).delete(int(params.dashboardId))

    if not result.success:
        raise Exception(f"Failed to delete dashboard: {result.error}")

    return ToolResult(content=[TextContent(text=json.dumps(result.data))])


def delete_dashboard_tool() -> Tool[DashboardDeleteSchema]:
    return Tool(
        name="dashboard-delete",
        description="Delete a dashboard by ID (soft delete - marks as deleted).",
        schema=DashboardDeleteSchema,
        handler=delete_dashboard_handler,
    )

File: typescript/src/tools/experiments/delete.ts
import { ExperimentDeleteSchema } from "@/schema/tool-inputs";
import { getToolDefinition } from "@/tools/toolDefinitions";
import type { Context, Tool } from "@/tools/types";
import type { z } from "zod";

const schema = ExperimentDeleteSchema;

type Params = z.infer<typeof schema>;

export const deleteHandler = async (context: Context, params: Params) => {
	const { experimentId } = params;
	const projectId = await context.stateManager.getProjectId();

	const deleteResult = await context.api.experiments({ projectId }).delete({
		experimentId,
	});
	if (!deleteResult.success) {
		throw new Error(`Failed to delete experiment: ${deleteResult.error.message}`);
	}

	return {
		content: [{ type: "text", text: JSON.stringify(deleteResult.data) }],
	};
};

const definition = getToolDefinition("experiment-delete");

const tool = (): Tool<typeof schema> => ({
	name: "experiment-delete",
	title: definition.title,
	description: definition.description,
	schema,
	handler: deleteHandler,
	annotations: {
		destructiveHint: true,
		idempotentHint: true,
		openWorldHint: true,
		readOnlyHint: false,
	},
});

export default tool;
